data_steward/cdr_cleaner/cleaning_rules/base_cleaning_rule.py

```diff
@@ -0,0 +1,248 @@
+                                        'affected_datasets')
+
+        # validate description is a string
+        self.__validate_string(self._description, 'description')
+
+        # validate project_id is a string
+        self.__validate_string(self._project_id, 'project_id')
+
+        # validate dataset_id is a string
+        self.__validate_string(self._dataset_id, 'dataset_id')
+
+        # validate sandbox_dataset_id is a string
+        self.__validate_string(self._sandbox_dataset_id, 'sandbox_dataset_id')
+
+        # depends_on_classes is allowed to be null,
+        # so is not validated against None
+        for clazz in self._depends_on_classes:
+            if not isinstance(clazz, BaseCleaningRule):
+                raise TypeError(
+                    '{} is expected to inherit from BaseCleaningRule'.format(
+                        self.__class__.__name__))
+
+    def get_depends_on_classes(self):
+        """
+        Return the names of classes the rule depends on.
+        """
+        return self._depends_on_classes
+
+    def get_jira_issue_numbers(self):
+        """
+        Return the jira_issue_numbers instance variable.
+        """
+        return self._jira_issue_numbers
+
+    def get_description(self):
+        """
+        Get the common language description of a cleaning rule.
+        """
+        return self._description
+
+    def get_affected_datasets(self):
+        """
+        Get the list of datasets a rule affects.
+        """
+        return self._affected_datasets
+
+    def get_project_id(self):
+        """
+        Get the project id for this class instance.
+        """
+        return self._project_id
+
+    def get_dataset_id(self):
+        """
+        Get the dataset id for this class instance.
+        """
+        return self._dataset_id
+
+    def get_sandbox_dataset_id(self):
+        """
+        Get the sandbox dataset id for this class instance.
+        """
+        return self._sandbox_dataset_id
+
+    def print_queries(self):
+        """
+        Helper function to print the SQL a class geenerates.
+
+        If the inheriting class builds tables inside the
+        get_query_dictionary_list function, the inheriting class will need
+        to override this function.
+        """
+        try:
+            query_list = self.get_query_dictionary_list()
+        except (oauth2client.client.HttpAccessTokenRefreshError,
+                googleapiclient.errors.HttpError, KeyError):
+            LOGGER.exception("Cannot list queries for %s",
+                             self.__class__.__name__)
+            raise
+
+        for query in query_list:
+            LOGGER.info('Generated SQL Query:\n%s',
+                        query.get(cdr_consts.QUERY, 'NO QUERY FOUND'))
+
+    @abstractmethod
+    def setup_query_execution(self, *args, **keyword_args):
+        """
+        Function to run any data upload options before executing a query.
```

Comment:
Make more explicit with something like: "Load resources (e.g. lookup tables) that may be required prior to executing the cleaning rule queries."